dexter/src/main/java/com/karumi/dexter/DexterInstance.java

```diff
@@ -285,20 +282,50 @@ private void checkSinglePermission(PermissionListener listener, String permissio
     checkMultiplePermissions(adapter, Collections.singleton(permission), thread);
   }
 
-  private void checkMultiplePermissions(MultiplePermissionsListener listener,
-      Collection<String> permissions, Thread thread) {
+  private void checkMultiplePermissions(final MultiplePermissionsListener listener,
+      final Collection<String> permissions, Thread thread) {
     checkNoDexterRequestOngoing();
     checkRequestSomePermission(permissions);
 
     pendingPermissions.clear();
     pendingPermissions.addAll(permissions);
     multiplePermissionsReport.clear();
     this.listener = new MultiplePermissionListenerThreadDecorator(listener, thread);
-
-    startTransparentActivityIfNeeded();
+    if (!everyPermissionIsGranted(permissions) || !contextIsInstanceOfActivity()) {
+      startTransparentActivityIfNeeded();
+    } else {
+      thread.execute(new Runnable() {
+        @Override public void run() {
+          MultiplePermissionsReport report = new MultiplePermissionsReport();
+          for (String permission : permissions) {
+            report.addGrantedPermissionResponse(PermissionGrantedResponse.from(permission));
+          }
+          isRequestingPermission.set(false);
+          listener.onPermissionsChecked(report);
+        }
+      });
+    }
     thread.loop();
   }
 
+  /**
+   * To be able to provide backward compatibility with the Dexter's 2.X.X version we need to check
+   * at some point if the context instance is an instance of {@link android.app.Activity}.
+   */
+  private boolean contextIsInstanceOfActivity() {
+    return context instanceof Activity;
+  }
+
+  private boolean everyPermissionIsGranted(Collection<String> permissions) {
```

Comment:
I'd call it `isEveryPermissionGranted` to follow java conventions (`is` prefix for things that return boolean values)